### PR TITLE
iox-#2047 Fix gcc 13 warnings

### LINKED ISSUE #2047: Support gcc 13.2?
## Brief feature description

Please describe in a few sentences what should be implemented

## Detailed information

E.g. a list of pros and cons of different considerations and how iceoryx and its user could benefit from it

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PR fixes the warnings in GCC 13. They are kind of false positives but I was able to give the compiler some hints to do not trigger them for some of the warnings.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #2047

### CODE REVIEW

**Comment by elBoberido** [comment]
@FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat @elfenpiff @budrus feel free to add yourself as reviewer

File: iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl [hunk 1]
```diff
@@ -140,7 +141,8 @@ template <typename ChunkDistributorDataType>
 inline uint64_t ChunkDistributor<ChunkDistributorDataType>::deliverToAllStoredQueues(mepoo::SharedChunk chunk) noexcept
 {
     uint64_t numberOfQueuesTheChunkWasDeliveredTo{0U};
-    typename ChunkDistributorDataType::QueueContainer_t remainingQueues;
+    using QueueContainer = decltype(getMembers()->m_queues);
+    QueueContainer fullQueuesAwaitingDelivery;
```
Comment (mossmaurice) [inline iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl:145]: :heart:

File: iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl [hunk 2]
```diff
@@ -171,43 +173,48 @@ inline uint64_t ChunkDistributor<ChunkDistributorDataType>::deliverToAllStoredQu
...
+            std::sort(fullQueuesAwaitingDelivery.begin(), fullQueuesAwaitingDelivery.end(), greaterThan);
+#if (defined(__GNUC__) && __GNUC__ == 13 && !defined(__clang__))
+#pragma GCC diagnostic pop
+#endif
+
+            std::set_intersection(getMembers()->m_queues.begin(),
+                                  getMembers()->m_queues.end(),
+                                  fullQueuesAwaitingDelivery.begin(),
+                                  fullQueuesAwaitingDelivery.end(),
+                                  std::back_inserter(remainingQueues),
```
Comment (mossmaurice) [inline iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl:204]: :rocket:

File: iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl [hunk 3]
```diff
@@ -171,43 +173,48 @@ inline uint64_t ChunkDistributor<ChunkDistributorDataType>::deliverToAllStoredQu
...
+            fullQueuesAwaitingDelivery.clear();
 
             // deliver to remaining queues
-            for (uint64_t i = remainingQueues.size() - 1U; !remainingQueues.empty(); --i)
+            for (auto& queue : remainingQueues)
             {
-                if (pushToQueue(remainingQueues[i].get(), chunk))
+                if (pushToQueue(queue.get(), chunk))
                 {
-                    remainingQueues.erase(remainingQueues.begin() + i);
```
Comment (mossmaurice) [inline iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl:203]: :heart:

**Review by mossmaurice — APPROVED** [review]
LGTM!

**Comment by elBoberido** [comment]
@mossmaurice I guess the eclipse admins did not yet applied the changes for only one required reviewer?

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**